src/api/course/index.ts

```diff
@@ -84,23 +94,137 @@ export class CourseApi {
       if (storeData) {
         this.emitCourseInfo(true)
       }
+      // register delta handler
+      ;(this.app as any).registerDeltaInputHandler(this.deltaHandler)
       resolve()
     })
   }
 
+  /** handle and process incoming course stream data */
+  private deltaHandler = (
+    delta: ValuesDelta,
+    next: (delta: ValuesDelta) => void
+  ) => {
+    this.processStreamSources(delta)
+    next(delta)
```

Comment:
If we always call delta this does this really need to be a deltahandler? And since there is no try-catch around processStreamSources then since this is a deltaHandler an error will result in delta being dropped.

This is just causing side-effect on certain path's values, right? Subscribing to those paths would make the code much more efficient, no need to compare each and every update's all pathvalues against those two paths.

Reply:
Try / Catch added.
Using a delta handler provides the option of filtering out / manipulating the original PathValue.
While it is not currently being done (to avoid breaking changes) I do foresee a need moving forward.

Reply:
Changhed to subscription.